### PR TITLE
Feat/implement life parameters

### PR DESCRIPTION
**Pull request by Wiecek-K**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Tamagotchi life parameters should decrease in time

### Requirements
1. `energy` should decrease by 1 point per 2 seconds
2. `hunger` should decrease by 1 points per second
3. `fun` should decrease by 1 points per second
4. `health` should decrease by 1 points per second when `hunger` or `energy` is below or equal to 0
5. `energy` should decrease by additional 1 point per 2 seconds when fun is below or equal to 0

### Code Review

#### public/styles.css
The changes in the CSS file are not related to the task requirements, but they seem to be fine, as they are adding a new class for styling.

#### src/app.ts
The changes in this file are in line with the task requirements. The game update function is called every second using `setInterval`, which will decrease the Tamagotchi's life parameters.

#### src/modules/abilities.ts
The class was changed from a default export to a named export. This change doesn't affect the task requirements.

#### src/modules/game.ts
The changes in this file are in line with the task requirements. The `update` method was added to the `Game` class, which calls the `decraseLifeParams` and `mount` methods of the `Tamagotchi` class every second.

#### src/modules/tamagotchi.ts
The changes in this file are mostly in line with the task requirements. The `decraseLifeParams` method was added to decrease the Tamagotchi's life parameters according to the rules specified in the task requirements. However, there is an issue with the initial value of `hunger`. According to the base parameter values provided in the task requirements, the initial value of `hunger` should be `10`, but in the changes, it was set to `5`.

### Suggestions
Change the initial value of `hunger` to `10` in the `Tamagotchi` class constructor:
```diff
- this.hunger = { value: 5, importance: 3 };
+ this.hunger = { value: 10, importance: 3 };
```

Overall, the code changes cover the task requirements, except for the initial value of `hunger`. After fixing this issue, the code should work as expected according to the task requirements.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Tamagotchi life parameters should decrease in time

### Requirements
1. `energy` should decrease by 1 point per 2 seconds
2. `hunger` should decrease by 1 points per second
3. `fun` should decrease by 1 points per second
4. `health` should decrease by 1 points per second when `hunger` or `energy` is below or equal to 0
5. `energy` should decrease by additional 1 point per 2 seconds when fun is below or equal to 0

### Code changes review

#### public/styles.css
The changes in the CSS file are not related to the task requirements, so they will not be reviewed.

#### src/app.ts
The changes in this file are correct. The `setInterval` function is set to run every second, which is needed for the requirements related to decreasing points per second.

#### src/modules/abilities.ts
The change in this file is not related to the task requirements, so it will not be reviewed.

#### src/modules/game.ts
The changes in this file are correct. The `update` method is added to the `Game` class, which calls the `decraseLifeParams` and `mount` methods of the `Tamagotchi` class. This is needed for the requirements related to decreasing points.

#### src/modules/tamagotchi.ts
The changes in this file are mostly correct. The `decraseLifeParams` method is added to the `Tamagotchi` class, which decreases the life parameters according to the task requirements. However, there's a potential issue with the way `energy` is decreased. According to the requirements, `energy` should decrease by 1 point per 2 seconds, and by an additional 1 point per 2 seconds when `fun` is below or equal to 0. But in the current implementation, `energy` is decreased by 1 point every second when `fun` is below or equal to 0. This does not meet the task requirements.

Here's a suggested change to meet the requirements:

```diff
decraseLifeParams = () => {
  if (this.hunger.value <= 0 || this.energy.value <= 0) {
    this.health.value--;
  }

- if (this.flag) {
-   this.energy.value--;
-   if (this.fun.value <= 0) {
-     this.energy.value--;
-   }
- } else {
-   this.flag = !this.flag;
- }
+ if (this.flag) {
+   if (this.fun.value <= 0) {
+     this.energy.value -= 2;
+   } else {
+     this.energy.value--;
+   }
+   this.flag = false;
+ } else {
+   this.flag = true;
+ }

  this.hunger.value--;
  this.fun.value--;
};
```

With this change, `energy` will decrease by 1 point per 2 seconds, and by an additional 1 point per 2 seconds when `fun` is below or equal to 0, which meets the task requirements.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Tamagotchi life parameters should decrease in time

### Requirements
1. `energy` should decrease by 1 point per 2 seconds
2. `hunger` should decrease by 1 points per second
3. `fun` should decrease by 1 points per second
4. `health` should decrease by 1 points per second when `hunger` or `energy` is below or equal to 0
5. `energy` should decrease by additional 1 point per 2 seconds when fun is below or equal to 0

### Code Changes Review

#### public/styles.css
The changes in this file are not directly related to the task requirements. However, they seem to be used for visual representation of the Tamagotchi's state.

#### src/app.ts
The changes in this file are consistent with the task requirements. The `setInterval` function is used to call the `game.update` method every second, which should decrease the Tamagotchi's life parameters according to the task requirements.

#### src/modules/abilities.ts
The change in this file is not directly related to the task requirements. The `Abilities` class is now exported without being a default export, which is a minor change and doesn't affect the functionality of the code.

#### src/modules/game.ts
The changes in this file are consistent with the task requirements. The `update` method was added to the `Game` class, which calls the `tamagotchi.decraseLifeParams` and `tamagotchi.mount` methods. This should decrease the Tamagotchi's life parameters and update their display every second.

#### src/modules/tamagotchi.ts
The changes in this file are mostly consistent with the task requirements. The `decraseLifeParams` method was added to the `Tamagotchi` class, which decreases the Tamagotchi's life parameters according to the task requirements. However, there is a potential issue with the `flag` variable, which is used to decrease the `energy` parameter every 2 seconds. Since the `decraseLifeParams` method is called every second, the `flag` variable will always alternate between `true` and `false`, which means that the `energy` parameter will decrease by 1 point every second, not every 2 seconds as required by the task. This issue can be fixed by using a counter instead of a boolean flag:

```diff
-  flag = false;
+  counter = 0;

  decraseLifeParams = () => {
    if (this.hunger.value <= 0 || this.energy.value <= 0) {
      this.health.value--;
    }

-    if (this.flag) {
+    if (this.counter % 2 === 0) {
      this.energy.value--;
      if (this.fun.value <= 0) {
        this.energy.value--;
      }
    }
-    this.flag = !this.flag;
+    this.counter++;
  };
```

In conclusion, the code changes mostly cover the task requirements, except for the issue with decreasing the `energy` parameter every 2 seconds.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
Tamagotchi life parameters should decrease in time

### Requirements
1. `energy` should decrease by 1 point per 2 seconds
2. `hunger` should decrease by 1 points per second
3. `fun` should decrease by 1 points per second
4. `health` should decrease by 1 points per second when `hunger` or `energy` is below or equal to 0
5. `energy` should decrease by additional 1 point per 2 seconds when fun is below or equal to 0

### Code Changes Review

#### public/styles.css
The changes in the CSS file are not related to the task requirements and seem to be more about visual representation of the Tamagotchi's state. 

#### src/app.ts
The changes in this file set up a game loop that updates every second, which is necessary for the Tamagotchi's life parameters to decrease over time. This meets the task requirements.

#### src/modules/abilities.ts
The change in this file is minor and doesn't affect the task requirements. It seems to be more about code organization.

#### src/modules/game.ts
The changes in this file add an update method to the game class, which decreases the Tamagotchi's life parameters and updates the display. This meets the task requirements.

#### src/modules/tamagotchi.ts
The changes in this file add methods to decrease the Tamagotchi's life parameters according to the rules specified in the task requirements. The `decraseLifeParams` method correctly implements all the rules. The changes also add methods to display the Tamagotchi's state, which is necessary for the player to know what's happening.

### Conclusion
The code changes cover all the task requirements. The Tamagotchi's life parameters decrease over time according to the rules specified in the task requirements, and the game updates every second. The changes also improve the code organization and the visual representation of the Tamagotchi's state.
